### DISCUSSION TITLE
Auto retreat

### DISCUSSION THREAD

**Discussion by AdriaGual** [AdriaGual/marvel-snap-bot#31] [Announcements]
So, @Wibbles790 had an idea to farm faster the season pass, the idea is simple, the bot will retreat the match after playing the 4rth turn, of course, this will make you lose cubes, but it will be optional to turn it on or not so, for the ones that like the original behaviour won't be affected.

Probably will make a new script, really similar to the `start.py` that we use, but with this tweak and another name like `start_auto_retreat.py`.

Another option is that I can add parameters to the python script, but that may be confusing for some people.

The original post is this issue, just for reference: [ISSUE](https://github.com/AdriaGual/marvel-snap-bot/issues/24)

What do you think?

**Comment by blamang** [comment]
That's a fantastic idea. I want to get it soon.

**Comment by zachleboeuf** [comment]
I did some testing a while back for farming season pass with Agatha. The amount of season pass points you get is equal to the number of rounds the game lasted + 10 x the number of locations you won.

Could someone please explain to me how forfeiting after turn 4 is a faster way to farm season pass experience?

**Comment by Neyschka** [comment]
Play a deck with lots of low cost cards. After 3 turns you'll be winning 2-3 locations 99% of the time. So when you forfeit on turn 4 you get 90%+ of the pass xp in about half the time of a 6 turn game. (emphasis on forfeiting on turn 4, not after turn 4)

**Comment by AdriaGual** [comment]
The auto retreat has been implemented!

Instead of running the `python start.py`, use the `python start_auto_concede.py`, it'll play normally till turn 4, when turn 4 is detected, it'll concede.

https://github.com/AdriaGual/marvel-snap-bot/commit/8423ad16fb798c86b9d0c4fcf7ea30df915b6874

**Reply by blamang** [reply]
Thank you so much!! I'm happy now

**Comment by ZloyPupsic** [comment]
got suspended, I think this mode is a reason. do not recommend to run it

**Reply by paperc07** [reply]
damn that sucks

**Reply by Nerdstrong** [reply]
how long was your suspension?
